js-pkg/packages/driftdb-react/src/index.tsx

```diff
@@ -13,59 +13,104 @@ import React, { useCallback, useEffect, useRef, useState } from 'react'
 
 const ROOM_ID_KEY = '_driftdb_room'
 
-export const DatabaseContext = React.createContext<DbConnection | null>(null)
-
-export function useDatabase(): DbConnection {
-  const db = React.useContext(DatabaseContext)
-  if (db === null) {
-    throw new Error('useDatabase must be used within a DriftDBProvider')
+/**
+ * A React component that provides a `DbConnection` to all child components.
+ * 
+ * @param props The props for the component.
+ */
+export function DriftDBProvider(props: {
+  /** Elements under the prodider in the tree. */
+  children: React.ReactNode
+  /** The URL of the DriftDB API. */
+  api: string
+  /** The room ID to connect to. If not provided, attempts to extract the room ID
+   *  from the URL and creates a new room if one is not present. */
+  room?: string
+  /** Whether to use binary messages (enables raw typed arrays in messages). */
+  useBinary?: boolean
+}): React.ReactElement {
+  const dbRef = useRef<DbConnection | null>(null)
+  if (dbRef.current === null) {
+    dbRef.current = new DbConnection()
   }
-  return db
-}
 
-export function RoomQRCode() {
-  const db = useDatabase()
-  const [pageUrl, setPageUrl] = React.useState<string | null>(null)
+  React.useEffect(() => {
+    let api = new Api(props.api)
 
-  useEffect(() => {
-    const callback = () => {
-      if (typeof window === 'undefined') {
-        return
-      }
+    let roomId
+    if (props.room) {
```

Comment:
Should we rerun this function whenever the `api`, `room`, and/or `useBinary` props change?

Reply:
Created https://github.com/drifting-in-space/driftdb/issues/74